### PR TITLE
feat: add expiration warning

### PR DESCRIPTION
**Pull request by rodrimaia**
Show a warning if your license is expiring soon:
image

### CODE REVIEW

File: enterprise/coderd/license/license_test.go [hunk 1]
```diff
@@ -102,6 +102,34 @@ func TestEntitlements(t *testing.T) {
 			fmt.Sprintf("%s is enabled but your license for this feature is expired.", codersdk.FeatureAuditLog.Humanize()),
 		)
 	})
+	t.Run("Expiration warning", func(t *testing.T) {
```
Comment (mtojek) [inline enterprise/coderd/license/license_test.go:105]: nit: maybe add one more test for the `1 day` case.

**Review by mtojek — APPROVED** [review]
LGTM, but please wait for 👍  from Ben

**Comment by mtojek** [comment]
@rodrimaia It looks like CI reported a problem [here](https://github.com/coder/coder/actions/runs/4787834849/jobs/8513615659?pr=7264):

```
    coderd_test.go:72: 
        	Error Trace:	/home/runner/work/coder/coder/enterprise/coderd/coderd_test.go:72
        	Error:      	Should be empty, but was [Your license expires in 1 day.]
        	Test:       	TestEntitlements/FullLicense
    stuntest.go:63: STUN server shutdown
```

File: enterprise/coderd/license/license.go [hunk 2]
```diff
@@ -70,6 +71,17 @@ func Entitlements(
 			// LicenseExpires we must be in grace period.
 			entitlement = codersdk.EntitlementGracePeriod
 		}
+
+		// Add warning if license is expiring soon
+		daysToExpire := int(math.Ceil(claims.LicenseExpires.Sub(now).Hours() / 24))
+		if daysToExpire > 0 && daysToExpire < 30 {
```
[the remark this answers is not in the document]
Comment (bpmct) [reply enterprise/coderd/license/license.go:77]: Let's do 30 days for non-trials and 7 days for trials. Does this message only show up for users who have privileges to add licenses to the deployment?

File: enterprise/coderd/license/license_test.go (hunk 1 above)
Comment (rodrimaia) [reply enterprise/coderd/license/license_test.go:105]: Fixed!

File: enterprise/coderd/license/license.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rodrimaia) [reply enterprise/coderd/license/license.go]: @bpmct cool! I've changed it to 30 days for non-trials and 7 for trials.

About the message, I could not find anything that prevents showing the warning for users without the privileges to add licenses. Not only for that specific warning but for all warnings and errors.  should we change it?

**Comment by rodrimaia** [comment]
> @rodrimaia It looks like CI reported a problem [here](https://github.com/coder/coder/actions/runs/4787834849/jobs/8513615659?pr=7264):
> 
> ```
>     coderd_test.go:72: 
>         	Error Trace:	/home/runner/work/coder/coder/enterprise/coderd/coderd_test.go:72
>         	Error:      	Should be empty, but was [Your license expires in 1 day.]
>         	Test:       	TestEntitlements/FullLicense
>     stuntest.go:63: STUN server shutdown
> ```

Fixed! :)

File: enterprise/coderd/license/license.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply enterprise/coderd/license/license.go]: I think so. @rodrimaia I think we can check for the same permission that we use to show or not the deployment settings. @bpmct does it make sense?

[the remark this answers is not in the document]
Comment (bpmct) [reply enterprise/coderd/license/license.go]: Yeah, I think we should only display this to admins who can change the license. Using the deployment settings makes sense

[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply enterprise/coderd/license/license.go]: Great!

**Pull request merged**